tfhe-zk-pok/src/curve_api.rs

```diff
@@ -329,12 +342,24 @@ impl CurveGroupOps<bls12_446::Zp> for bls12_446::G1 {
     }
 
     fn mul_scalar(self, scalar: bls12_446::Zp) -> Self {
-        self.mul_scalar(scalar)
+        if scalar.inner == MontFp!("2") {
+            self.double()
+        } else {
+            self.mul_scalar(scalar)
+        }
     }
 
+    #[track_caller]
     fn multi_mul_scalar(bases: &[Self::Affine], scalars: &[bls12_446::Zp]) -> Self {
-        msm::msm_wnaf_g1_446(bases, scalars)
-        // Self::Affine::multi_mul_scalar(bases, scalars)
+        // overhead seems to not be worth it outside of wasm
+        #[cfg(target_family = "wasm")]
+        {
+            msm::msm_wnaf_g1_446(bases, scalars)
+        }
+        #[cfg(not(target_family = "wasm"))]
+        {
+            Self::Affine::multi_mul_scalar(bases, scalars)
+        }
```

Comment:
is the `if cfg!(target_family = "wasm") {` usable here ? I think it makes it easier to read

Reply:
yeah

Reply:
fixed